plugin/trino-postgresql/src/test/java/io/trino/plugin/postgresql/TestPostgreSqlJdbcConnectionCreation.java

```diff
@@ -0,0 +1,181 @@
+        try {
+            queryRunner = DistributedQueryRunner.builder(createSession())
+                    .build();
+
+            queryRunner.installPlugin(new TpchPlugin());
+            queryRunner.createCatalog("tpch", "tpch");
+
+            Map<String, String> connectorProperties = new HashMap<>();
+            connectorProperties.putIfAbsent("connection-url", server.getJdbcUrl());
+            connectorProperties.putIfAbsent("connection-user", server.getUser());
+            connectorProperties.putIfAbsent("connection-password", server.getPassword());
+            connectorProperties.putIfAbsent("postgresql.include-system-tables", "true");
+
+            queryRunner.installPlugin(new TestingPostgreSqlPlugin(connectionCountingConnectionFactory));
+            queryRunner.createCatalog("postgresql", "postgresql", connectorProperties);
+
+            copyTpchTables(queryRunner, "tpch", TINY_SCHEMA_NAME, createSession(), tables);
+
+            return queryRunner;
+        }
+        catch (Throwable e) {
+            closeAllSuppress(e, queryRunner, server);
+            throw e;
+        }
+    }
+
+    private static class TestingPostgreSqlModule
+            extends AbstractConfigurationAwareModule
+    {
+        private final ConnectionCountingConnectionFactory connectionCountingConnectionFactory;
+
+        TestingPostgreSqlModule(ConnectionCountingConnectionFactory connectionCountingConnectionFactory)
+        {
+            this.connectionCountingConnectionFactory = requireNonNull(connectionCountingConnectionFactory, "connectionCountingConnectionFactory is null");
+        }
+
+        @Override
+        protected void setup(Binder binder) {}
+
+        @Provides
+        @Singleton
+        @ForBaseJdbc
+        public ConnectionFactory getConnectionFactory(BaseJdbcConfig config, CredentialProvider credentialProvider)
+        {
+            return connectionCountingConnectionFactory;
+        }
+    }
+
+    private static class TestingPostgreSqlPlugin
+            implements Plugin
+    {
+        private final ConnectionCountingConnectionFactory connectionCountingConnectionFactory;
+
+        public TestingPostgreSqlPlugin(ConnectionCountingConnectionFactory connectionCountingConnectionFactory)
+        {
+            this.connectionCountingConnectionFactory = requireNonNull(connectionCountingConnectionFactory, "connectionCountingConnectionFactory is null");
+        }
+
+        @Override
+        public Iterable<ConnectorFactory> getConnectorFactories()
+        {
+            return ImmutableList.of(new JdbcConnectorFactory(
+                    "postgresql",
+                    combine(
+                            new PostgreSqlClientModule(),
+                            new TestingPostgreSqlModule(connectionCountingConnectionFactory),
+                            new ExtraCredentialsBasedIdentityCacheMappingModule(),
+                            new CredentialProviderModule())));
```

Comment:
Can this be simpler?
E.g. we don't have any extra credentials in the test, and also a StaticCredentialProvider was already constructed earlier.

Reply:
I would prefer this testing plugin for postgresql to be as close to real as possible. We don't win much using things like `StaticCredentialProvider` and we may end up having completely two connectors for postgresql, which I would like to avoid.